povm_toolbox/library/dilation_measurement.py

```diff
@@ -0,0 +1,299 @@
+
+        composed_circuit = self.compose_circuits(circuit)
+
+        if pass_manager is not None:
+            composed_circuit = pass_manager.run(composed_circuit)
+
+        pub = SamplerPub(
+            circuit=composed_circuit,
+            parameter_values=circuit_binding,
+            shots=shots,
+        )
+
+        metadata = POVMMetadata(povm_implementation=self, composed_circuit=composed_circuit)
+
+        t2 = time.time()
+        LOGGER.info(f"Finished building SamplerPub. Took {t2 - t1:.6f}s")
+
+        return (pub, metadata)
+
+    @override
+    def reshape_data_bin(self, data: DataBin) -> DataBin:
+        return data
+
+    @override
+    def _povm_outcomes(
+        self,
+        bit_array: BitArray,
+        povm_metadata: POVMMetadata,
+        loc: int | tuple[int, ...] | None = None,
+    ) -> list[tuple[int, ...]]:
+        t1 = time.time()
+        LOGGER.info("Creating POVM outcomes")
+
+        # loc is assumed to have a length of at most pv.ndim = len(pv.shape)
+
+        povm_outcomes = [
+            self._get_outcome_label(bitstring_outcome=raw_bitstring)
+            for raw_bitstring in bit_array.get_bitstrings(loc)
+        ]
+
+        t2 = time.time()
+        LOGGER.info(f"Finished creating POVM outcomes. Took {t2 - t1:.6f}s")
+
+        return povm_outcomes
+
+    def _get_outcome_label(
+        self,
+        bitstring_outcome: str,
+    ) -> tuple[int, ...]:
+        """Transform a bitstring outcome to a POVM outcome.
+
+        Args:
+            bitstring_outcome: the raw outcome of the measurement. The order of
+            qubit is assumed to be reversed. More specifically, the first
+            :attr:`.num_qubits` bits will correspond to the ancilla qubit (in
+            reverse order) and the last :attr:`.num_qubits` bits will correspond
+            to the qubit to measure (in reverse order).
+
+        Returns:
+            A tuple of indices indicating the POVM outcomes on each qubit. For each qubit,
+            the index goes from :math:``0`` to :math:``3``.
+        """
+        return tuple(
+            2 * int(bit_q) + int(bit_a)
+            for bit_q, bit_a in zip(
+                bitstring_outcome[: self.num_qubits - 1 : -1],
+                bitstring_outcome[self.num_qubits - 1 :: -1],
+            )
+        )
+
+    def _unitary_from_parameters(self, param: np.ndarray):
+        """Construct the unitary defining the dilation POVM from parameters.
+
+        We use the same parametrization of the dilation POVM as the one presented in the
+        work of G. García-Pérez, M. A. Rossi, B. Sokolov, F. Tacchino, P. K. Barkoutsos,
+        G. Mazzola, I. Tavernelli, and S. Maniscalco, “*Learning to measure: adaptive
+        informationally complete generalized measurements for quantum algorithms*”, PRX
+        Quantum 2, Publisher: American Physical Society, 040342 (2021). Refer to this work
+        for a detailed explanation of the parametrization.
```

Comment:
I don't think we have to duplicate this from the main docstring. You can simply refer to it if you want to add an explicit link.
But leaving that implied from this internal method should be fine, too